client/driver/docker.go

```diff
@@ -1302,74 +1302,73 @@ CREATE:
 
 	d.logger.Printf("[DEBUG] driver.docker: failed to create container %q from image %q (ID: %q) (attempt %d): %v",
 		config.Name, d.driverConfig.ImageName, d.imageID, attempted+1, createErr)
-	if strings.Contains(strings.ToLower(createErr.Error()), "container already exists") {
-		containers, err := client.ListContainers(docker.ListContainersOptions{
-			All: true,
-		})
-		if err != nil {
-			d.logger.Printf("[ERR] driver.docker: failed to query list of containers matching name:%s", config.Name)
-			return nil, recoverableErrTimeouts(fmt.Errorf("Failed to query list of containers: %s", err))
-		}
-
-		// Delete matching containers
-		// Adding a / infront of the container name since Docker returns the
-		// container names with a / pre-pended to the Nomad generated container names
-		containerName := "/" + config.Name
-		d.logger.Printf("[DEBUG] driver.docker: searching for container name %q to purge", containerName)
-		for _, shimContainer := range containers {
-			d.logger.Printf("[DEBUG] driver.docker: listed container %+v", shimContainer.Names)
-			found := false
-			for _, name := range shimContainer.Names {
-				if name == containerName {
-					d.logger.Printf("[DEBUG] driver.docker: Found container %v: %v", containerName, shimContainer.ID)
-					found = true
-					break
-				}
-			}
+	if strings.Contains(strings.ToLower(createErr.Error()), "no such image") {
+		// There is still a very small chance this is possible even with the
+		// coordinator so retry.
+		return nil, structs.NewRecoverableError(createErr, true)
+	}
 
-			if !found {
-				continue
+	containers, err := client.ListContainers(docker.ListContainersOptions{
+		All: true,
+	})
+	if err != nil {
+		d.logger.Printf("[ERR] driver.docker: failed to query list of containers matching name:%s", config.Name)
+		return nil, recoverableErrTimeouts(fmt.Errorf("Failed to query list of containers: %s", err))
+	}
+
+	// Delete matching containers
+	// Adding a / infront of the container name since Docker returns the
+	// container names with a / pre-pended to the Nomad generated container names
+	containerName := "/" + config.Name
+	d.logger.Printf("[DEBUG] driver.docker: searching for container name %q to purge", containerName)
```

Comment:
Can you remove `to purge` from this log line? I know you didn't add it, but it seems grossly misleading as if we find a running container we do *not* purge it.